recourse_methods/dice_method.py

```diff
@@ -0,0 +1,108 @@
+from typing import Sequence, Optional, Mapping, Any
+from recourse_methods import base_type
+import pandas as pd
+from data import data_preprocessor as dp
+import dice_ml
+from dice_ml import constants
+from sklearn import pipeline
+
+
+class ToNumpy:
+    def __init__(self):
+        pass
+
+    def fit(self, X, y):
+        pass
+
+    def transform(self, X):
+        return X.to_numpy()
+
+
+class DiCE(base_type.RecourseMethod):
+    """An abstract base class for recourse methods."""
+
+    def __init__(self,
+                 k_directions: int,
+                 preprocessor: dp.Preprocessor,
+                 dataset: pd.DataFrame,
+                 continuous_features: Sequence[str],
+                 model: Any,
+                 model_backend: constants.BackEndTypes,
+                 label_column: str = 'Y',
+                 dice_kwargs: Optional[Mapping[str, Any]] = None,
+                 dice_cfe_kwargs: Optional[Mapping[str, Any]] = None):
+        """Constructs a new DiCE recourse method.
+        
+        Args:
+            k_directions: The number of recourse directions to generate.
+            preprocessor: The dataset preprocessor.
+            dataset: The dataset to perform recourse over.
+            continuous_features: A list of the dataset's continuous features.
+            model: An ML model satisfying one of the DiCE model backends.
+            model_backend: Model types recognized by DiCE (sklearn, pytorch, and tensorflow).
+            label_column: The column containing binary classification outputs.
+            dice_kwargs: Optional arguments to pass to DiCE on instantiation.
+            dice_recourse_kwargs: Optional arguments to pass to DiCE on counterfactual explanation generation.
+        """
```

Comment:
Are there any likely failure modes or exceptions that may need handling? If so, you can add a note here describing that/those.